### PR TITLE
Unify blob store compress setting

### PR DESCRIPTION
**Pull request by henningandersen**
Blob store compression was all implemented generally, except reading the
setting for it. Moved the setting to BlobStoreRepository to unify this.
Also removed deprecated env setting 'repositories.fs.compress'.

This is a follow up on #39073

Given this is not documented, does this need to be included in the breaking changes
documentation for 8.0?

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -248,40 +247,28 @@ public void testBadChunksize() throws Exception {
...
-            .put(node().getEnvironment().settings())
-            .put(FsRepository.REPOSITORIES_COMPRESS_SETTING.getKey(), true)
-            .build();
-        Environment useCompressEnvironment =
-            new Environment(useCompressSettings, node().getEnvironment().configFile());
-
-        new FsRepository(metaData, useCompressEnvironment, null);
-
-        assertWarnings("[repositories.fs.compress] setting was deprecated in Elasticsearch and will be removed in a future release!" +
-            " See the breaking changes documentation for the next major version.");
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java]: @henningandersen maybe we should add a small note, given that we kept logging this?

**Review by original-brownbear — APPROVED** [review]
LGTM :)

Not sure about adding something to the breaking changes (instinctively it may be a good idea, given the previous log message) => @ywelsch wdyt?

**Review by ywelsch — APPROVED** [review]
LGTM. I'm inclined to add a note to the breaking changes docs, as I'm not sure exactly for what purpose this was added and who might be using it. @imotov @dadoonet do you remember where this undocumented node setting `repositories.fs.compress` is coming from?

File: server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java (hunk 1 above)
Comment (henningandersen) [reply server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java]: Yes, makes sense, thanks.

**Comment by henningandersen** [comment]
I added the breaking changes documentation for the removed setting.

**Review by ywelsch — APPROVED** [review]

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run CLA

**Pull request merged**